### DISCUSSION TITLE
Service account object must contain a string \"private_key\" property."}

### DISCUSSION THREAD

**Discussion by 6174** [gladly-team/next-firebase-auth#499] [Q&A] [answered]
**Describe the bug**
New to next.js & firebase, my code is the same with example, but stuck on this issue, I find the same issue in the issue list, but no solution. 
![image](https://user-images.githubusercontent.com/3872872/180206326-9e9b88bb-5b82-4502-93a6-6767c4ab49d6.png)

the code
```ts
firebaseAdminInitConfig: {
      credential: {
        privateKey: process.env.FIREBASE_PRIVATE_KEY
          ? JSON.parse(process.env.FIREBASE_PRIVATE_KEY)
          : undefined,
      },
    },
```

**Versions**

`next-firebase-auth` version: 
Firebase JS SDK: ^9.9.0
Next.js: 12.2.2

**Additional context**
I  log out the private key with JSON.parse
```
pk {
  type: 'service_account',
  project_id: 'xxxx',
  private_key_id: 'xxxx',
  private_key: '-----BEGIN PRIVATE KEY-----\n' +
    'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' +
    'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' +
    'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' +
```

**Comment by kmjennison** [comment, marked as answer]
You should confirm that the `credential.privateKey` value is only the private key, not the whole service account object. In local development, you could try hardcoding the private key in the `next-firebase-auth` config rather than using environment variables at all—just to make sure it works.

Related, this thread includes suggestions on formatting the private key:
https://github.com/gladly-team/next-firebase-auth/discussions/95

**Reply by 6174** [reply]
Thx, change the "whole service account object" to key string fixed the issue.
